torch/quantization/_quantize_script.py

```diff
@@ -82,13 +92,23 @@ def get_scripted_qconfig_dict(qconfig_dict):
 
 def _prepare_script(model, qconfig_dict, is_dynamic):
     _check_is_script_module(model)
+    if not all(map(lambda x : isinstance(x, str), qconfig_dict.keys())):
+        raise ValueError('qconfig_dict should contain names(str) as keys. '
+                         'Run preprocess_qconfig_dict first to convert types to names')
     scripted_qconfig_dict = get_scripted_qconfig_dict(qconfig_dict)
     return wrap_cpp_module(torch._C._jit_pass_insert_observers(model._c,
                                                                'forward',
                                                                scripted_qconfig_dict,
                                                                False,
                                                                is_dynamic))
 
+def preprocess_qconfig_dict(module, qconfig_dict):
+    dict_has_types = any(map(lambda x : isinstance(x, type), qconfig_dict.keys()))
+    if dict_has_types:
+        qconfig_dict = _preprocess_qconfig_dict_helper(module, qconfig_dict)
```

Comment:
we can copy the qconfig_dict here and make a inplace modification in the function.